providers/aws/services/autoscaling/autoscaling_service_test.py

```diff
@@ -0,0 +1,91 @@
+class Test_AutoScaling_Service:
+    # Mocked Audit Info
+    def set_mocked_audit_info(self):
+        audit_info = AWS_Audit_Info(
+            original_session=None,
+            audit_session=session.Session(
+                profile_name=None,
+                botocore_session=None,
+            ),
+            audited_account=AWS_ACCOUNT_NUMBER,
+            audited_user_id=None,
+            audited_partition="aws",
+            audited_identity_arn=None,
+            profile=None,
+            profile_region=None,
+            credentials=None,
+            assumed_role_info=None,
+            audited_regions=None,
+            organizations_metadata=None,
+        )
+        return audit_info
+
+    # Test AutoScaling Service
+    @mock_autoscaling
+    def test_service(self):
+        # AutoScaling client for this test class
+        audit_info = self.set_mocked_audit_info()
+        autoscaling = AutoScaling(audit_info)
+        assert autoscaling.service == "autoscaling"
+
+    # Test AutoScaling Client
+    @mock_autoscaling
+    def test_client(self):
+        # AutoScaling client for this test class
+        audit_info = self.set_mocked_audit_info()
+        autoscaling = AutoScaling(audit_info)
+        for client in autoscaling.regional_clients.values():
+            assert client.__class__.__name__ == "AutoScaling"
+
+    # Test AutoScaling Session
+    @mock_autoscaling
+    def test__get_session__(self):
+        # AutoScaling client for this test class
+        audit_info = self.set_mocked_audit_info()
+        autoscaling = AutoScaling(audit_info)
+        assert autoscaling.session.__class__.__name__ == "Session"
+
+    # Test AutoScaling Session
+    @mock_autoscaling
+    def test_audited_account(self):
+        # AutoScaling client for this test class
+        audit_info = self.set_mocked_audit_info()
+        autoscaling = AutoScaling(audit_info)
+        assert autoscaling.audited_account == AWS_ACCOUNT_NUMBER
+
+    # Test AutoScaling Get APIs
+    @mock_autoscaling
+    def test__describe_launch_configurations__(self):
+        # Generate AutoScaling Client
+        autoscaling_client = client("autoscaling", region_name=AWS_REGION)
+        # Create AutoScaling API
+        autoscaling_client.create_launch_configuration(
+            LaunchConfigurationName="tester1",
+            ImageId="ami-12c6146b",
+            InstanceType="t1.micro",
+            KeyName="the_keys",
+            SecurityGroups=["default", "default2"],
+        )
+        autoscaling_client.create_launch_configuration(
+            LaunchConfigurationName="tester2",
+            ImageId="ami-12c6146b",
+            InstanceType="t1.micro",
+            KeyName="the_keys",
+            SecurityGroups=["default", "default2"],
+        )
+        # AutoScaling client for this test class
+        audit_info = self.set_mocked_audit_info()
+        autoscaling = AutoScaling(audit_info)
+        assert len(autoscaling.launch_configurations) == len(
```

Comment:
Please, assert all the values present in the launch_configurations object.

Reply:
Yes, but you can check the values within the LaunchConfiguration object and also create one with a UserData and validate the content.

Reply:
Done